R/checks.R

```diff
@@ -97,14 +96,12 @@ check_repeat_and_nonrepeat <- function(db_data) {
   # Step (3)
   repeat_nonrepeat_error <- function(check_data, names){
 
+    rep <- gsub(pattern = "_repeatingcheck", replacement = "", x = names)
+
     if ("repeating" %in% check_data &
         "nonrepeating" %in% check_data){
-      stop(
-        paste0("Instrument detected that has both repeated and nonrepeated
-               instances defined in the project: ",
-               gsub(pattern = "_repeatingcheck", replacement = "", x = names)
-        )
-      )
+      cli_abort(c("x" = "Instrument detected that has both repeated and
```

Comment:
I think for most error conditions we should create a specific named error condition class. And there should be a test that makes sure that specific error condition is being caught (and not some other error).

Reply:
Agree on that. I think I roughly did that here for example:
https://github.com/CHOP-CGTDataOps/REDCapTidieR/blob/500b53f9497cd8ee61edce66852812f1d882dd28/R/utils.R#L339-L342

We may want to create a couple utility functions like `custom_error()` and `custom_warn()` so we can update the class in one place if we ever needed to. Like:

```r
custom_warn <- function(msg, class = NULL) {
    cli_warn(c("!" = msg), class = c(class, "REDCapTidieR_cond"))
}
```

Reply:
For now I will implement a similar `class` to Ezra's, I like having a common "REDCapTidieR_cond" as one of them and a more specific one. If we want to make the custom error func happy to implement, for now I will apply the classes. 👍